src/csharp/Microsoft.Spark/Sql/RowConstructor.cs

```diff
@@ -11,105 +11,48 @@
 namespace Microsoft.Spark.Sql
 {
     /// <summary>
-    /// RowConstructor is a custom unpickler for GenericRowWithSchema in Spark.
+    /// RowConstructorConstructor is a custom unpickler for GenericRowWithSchema in Spark.
     /// Refer to spark/sql/core/src/main/scala/org/apache/spark/sql/execution/python/
     /// EvaluatePython.scala how GenericRowWithSchema is being pickled.
     /// </summary>
-    internal sealed class RowConstructor : IObjectConstructor
+    internal sealed class RowConstructorConstructor : IObjectConstructor
```

Comment:
Is there a better name than `RowConstructorConstructor` or is this a convention?

Reply:
I'm okay with any name suggestion.  `RowConstructorFactory`?  Rename back to `RowConstructor` and change the class at the bottom to `RowWithSchemaConstructor` ?

Reply:
the latter sounds fine to me?

Reply:
renamed to latter.